scripts/firedrake-install

```diff
@@ -299,21 +303,34 @@ def git_clone(url):
             raise Exception
         check_call(["git", "clone", "-q", git_url(plain_url, "ssh")])
         log.info("Successfully cloned repository %s" % name)
-        with directory(name):
-            check_call(["git", "checkout", "-q", branch])
-            log.info("Successfully checked out branch %s" % branch)
     except:
         if not args.disable_ssh:
             log.warn("Failed to clone %s using ssh, falling back to https." % name)
         try:
             check_call(["git", "clone", "-q", git_url(plain_url, "https")])
             log.info("Successfully cloned repository %s." % name)
-            with directory(name):
-                check_call(["git", "checkout", "-q", branch])
-                log.info("Successfully checked out branch %s" % branch)
         except subprocess.CalledProcessError:
             log.error("Failed to clone %s branch %s." % (name, branch))
             raise
+    with directory(name):
+        if name == "firedrake" and ci and ci_event == "pull_request":
+            try:
+                ci_pull_request = os.environ["CI_PULL_REQUEST"]
+                log.info("Checking out pull request %s" % ci_pull_request)
+                check_call(["git", "fetch", "origin", "pull/" + ci_pull_request + "/merge"])
+                check_call(["git", "checkout", "FETCH_HEAD"])
+                log.info("Successfully checked out pull request %s" % ci_pull_request)
+            except subprocess.CalledProcessError:
+                log.error("Failed to check out pull request %s" % ci_pull_request)
+                raise
+        else:
+            try:
+                log.info("Checking out branch %s" % branch)
+                check_call(["git", "checkout", "-q", branch])
+                log.info("Successfully checked out branch %s" % branch)
+            except subprocess.CalledProcessError:
+                log.error("Failed to check out branch %s" % branch)
+                raise
```

Comment:
I note that the travis hack for this happens in `git_update`, not `git_clone`.  Any reason for the difference?